tests/app/dao/test_template_categories_dao.py

```diff
@@ -0,0 +1,368 @@
+            {
+                "name_en": "english",
+                "name_fr": "french",
+                "description_en": "english description",
+                "description_fr": "french description",
+                "sms_process_type": NORMAL,
+                "email_process_type": NORMAL,
+                "hidden": False,
+            },
+            {
+                "name_en": "new english",
+                "name_fr": "new french",
+                "description_en": "new english description",
+                "description_fr": "new french description",
+                "sms_process_type": BULK,
+                "email_process_type": BULK,
+                "hidden": True,
+            },
+        )
+    ],
+)
+def test_update_template_category(notify_db_session, category, updated_category):
+    template_category = TemplateCategory(**category)
+    dao_create_template_category(template_category)
+
+    for key, value in updated_category.items():
+        setattr(template_category, key, value)
+
+    dao_update_template_category(template_category)
+
+    fetched_category = dao_get_all_template_categories()[0]
+
+    assert fetched_category.id == template_category.id
+    for key, value in updated_category.items():
+        assert getattr(fetched_category, key) == value
+
+
+@pytest.mark.parametrize(
+    "category, template",
+    [
+        (
+            {
+                "name_en": "english",
+                "name_fr": "french",
+                "description_en": "english description",
+                "description_fr": "french description",
+                "sms_process_type": NORMAL,
+                "email_process_type": NORMAL,
+                "hidden": False,
+            },
+            {
+                "name": "Sample Template",
+                "template_type": "email",
+                "content": "Template content",
+            },
+        )
+    ],
+)
+def test_dao_get_template_category_by_template_id(category, template, notify_db_session, sample_service, sample_user):
+    template_category = TemplateCategory(**category)
+    dao_create_template_category(template_category)
+
+    template = Template(**template)
+    template.service = sample_service
+    template.created_by = sample_user
+    template.category = template_category
+    dao_create_template(template)
+
+    assert dao_get_template_category_by_template_id(template.id) == template_category
+
+
+def test_get_template_category_by_id(notify_db_session):
+    data = {
+        "name_en": "english",
+        "name_fr": "french",
+        "description_en": "english description",
+        "description_fr": "french description",
+        "sms_process_type": NORMAL,
+        "email_process_type": NORMAL,
+        "hidden": False,
+    }
+
+    template_category = TemplateCategory(**data)
+    dao_create_template_category(template_category)
+
+    assert dao_get_template_category_by_id(template_category.id) == template_category
+
+
+@pytest.mark.parametrize(
```

Comment:
Originally I wanted to cut down the size of these parameters by creating the `categories_to_insert` in the test body and modifying the `hidden` field per test. When I did it this way, then between the tests the categories were not being removed from the test DB and the unique constraint on the `name_en` and `name_fr` was causing the tests to fail. 

I can leave it as is or do something like set the name to a `str(uuid)` to get around the unique constraint and shrink the size of this behemoth.